test/data/meson.build

```diff
@@ -1,5 +1,6 @@
 data_files = [
   'small.zim',
+  'small.zim.embedded',
```

Comment:
(Not specific to this line, but I have to put the comment somewhere)

Why this commit changes all test zim files ?
It should only add a new zim file.

Reply:
That's an issue with the `create_test_zimfiles` script. It always regenerates all test zimfiles and the new ones don't bitwise match the old ones even if no source data has changed. I usually committed new test zim files selectively. On a couple of occasions (though not in this PR) I also fixed some bugs in the `create_test_zimfiles` script and I decided to re-commit all  ZIM files so that they form a coherent set. I don't remember exactly why I chose to do the same in this commit, but logic suggests that `small.zim.embedded` must be coherent with `small.zim` and the rest (broken ZIM files) were committed to be coherent with the source file from which they were produced.

Reply:
> the new ones don't bitwise match the old ones even if no source data has changed

This should not be the case.
There is no reason output changes if input or script don't.

We have more than 1Mb of test zim file.
If we change them every time we add a new test we will pollute our git repository with binary data.

We should maybe move them to git lfs, Or at least only keep only the source zim file and generate other test file at runtime.